src/devices/cpu/z180/z180asci.h

```diff
@@ -0,0 +1,153 @@
+// license:BSD-3-Clause
+// copyright-holders:Miodrag Milanovic
+/*********************************************************************
+
+    z180asci.h
+
+*********************************************************************/
+
+#ifndef MAME_CPU_Z180_Z180ASCI_H
+#define MAME_CPU_Z180_Z180ASCI_H
+
+#pragma once
+
+#include "diserial.h"
+
+//**************************************************************************
+//  z180asci_channel_base
+//**************************************************************************
+
+class z180asci_channel_base : public device_t, public device_buffered_serial_interface<4U>
```

Comment:
Where are you getting the four-deep FIFO from?  From a quick read of the z8018x family manual, my takeaway was that it only has a single transmit byte register in addition to the transmit shift register, much like other simple U(S)ARTs.

Reply:
This is part from documentation:
```
ASCI Receive Data FIFO 0,1 (RDR0, RDR1)—I/O Address = 08h, 09h. The ASCI
Receive Data Register is a READ-ONLY register. When a complete incoming data byte is
assembled in RSR, it is automatically transferred to the 4 character Receive Data First-In
First-Out (FIFO) memory. The oldest character in the FIFO (if any) can be read from the
Receive Data Register (RDR). The next incoming data byte can be shifted into RSR while
the FIFO is full. The ASCI receiver is well buffered.
```

Reply:
Yes, but the FIFO in `device_buffered_serial_interface` is on the *transmission* side.  It dosn’t have a receive FIFO.  The class is quite short, and you can see that the FIFO is not involved in reception.